src/main.ts

```diff
@@ -545,6 +546,30 @@ export async function getToken(input: CspInput): Promise<string> {
   }
 }
 
+export async function checkTokenExpiration(input: CspInput): Promise<string> {
+  const response = await cspClient.post("https://console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/details", {
+    headers: {
+      "Content-Type": "application/json",
+      tokenValue: "$CSP_API_TOKEN",
+    },
+  })
+
+  cachedCspToken = {
+    access_token: response.data.access_token,
+    timestamp: Date.now() + input.timeout,
+  }
+
+  if (response.data.details) {
+    return response.data.expiresAt
+  }
+
+  if (response.data.expiresAt < `${moment().add(1, "month").calendar()}`) {
+    core.warning(`CSP API token will expire in ${moment().add(1, "month").calendar()}.`)
+  }
+
+  return response.data.details
```

Comment:
You need to be consistent in the return value. Above, you're returning `expireAt` and here you are returning `details`.